src/easier68k/core/opcodes/lea.py

```diff
@@ -0,0 +1,204 @@
+        dest = parse_assembly_parameter(params[1].strip())
+
+        return cls(src, dest), issues
+
+    def __init__(self, src: EAMode, dest: EAMode):
+        # Can't take data register, address register, or ARI with modifications
+        assert src.mode not in [EAMode.DRD, EAMode.ARD, EAMode.ARIPD, EAMode.ARIPI]
+        self.src = src
+
+        # Check that the destination is of a proper type
+        assert dest.mode == EAMode.ARD  # Can only take address register direct
+        self.dest = dest
+
+    def assemble(self) -> bytearray:
+        """
+        Assembles this opcode into hex to be inserted into memory
+        :return: The hex version of this opcode
+        """
+        # Create a binary string to append to, which we'll convert to hex at the end
+        tr = '0100'  # Opcode
+        tr += '{0:03b}'.format(self.dest.data)
+        tr += '111'
+        tr += ea_mode_bin.parse_from_ea_mode_modefirst(self.src)  # Source second
+        # Append immediates/absolute addresses after the command
+        if self.src.mode is EAMode.IMM:
+            if len(hex(self.src.data)[2:]) > 4:
+                tr += opcode_util.ea_to_binary_post_op(self.src, 'L')
+            else:
+                tr += opcode_util.ea_to_binary_post_op(self.src, 'W')
+        else: # Size doesn't matter if it's not an immediate so we'll just give it W
+            tr += opcode_util.ea_to_binary_post_op(self.src, 'W')
+
+        to_return = bytearray.fromhex(hex(int(tr, 2))[2:])  # Convert to a bytearray
+        return to_return
+
+    def execute(self, simulator: M68K):
+        """
+        Executes this command in a simulator
+        :param simulator: The simulator to execute the command on
+        :return: Nothing
+        """
+        pass
+
+    def __str__(self):
+        # Makes this a bit easier to read in doctest output
+        return 'LEA command: src {}, dest {}'.format(self.src, self.dest)
+
+    @staticmethod
+    def is_valid(command: str, parameters: str) -> (bool, list):
+        """
+        Tests whether the given command is valid
+
+        >>> Lea.is_valid('LEA', '(A0), A1')[0]
+        True
+
+        >>> Lea.is_valid('LEA', 'A0')[0]
+        False
+
+        >>> Lea.is_valid('LEA.B', '(A0), A1')[0]
+        False
+
+        >>> Lea.is_valid('LEA', 'D0, A2')[0]
+        False
+
+        >>> Lea.is_valid('LEA', '#$0A, A4')[0]
+        True
+
+        >>> Lea.is_valid('LEA', '($AAAA).L, A7')[0]
+        True
+
+        :param command: The command itself (e.g. 'MOVE.B', 'LEA', etc.)
+        :param parameters: The parameters after the command (such as the source and destination of a move)
+        :return: Whether the given command is valid and a list of issues/warnings encountered
+        """
+        issues = []
+        try:
+            assert opcode_util.check_valid_command(command, 'LEA', can_take_size=False), 'Command invalid'
+
+            # Split the parameters into EA modes
+            params = parameters.split(',')
```

Comment:
Why not use that utility function?

Or, was that in the PR I'm working on that hasn't been merged yet?

Reply:
That hasn't been merged yet/this isn't rebased yet where that exists (one way or another it's not here yet but will be soon)